SensusUI/PromptForInputsPage.cs

```diff
@@ -83,6 +83,24 @@ public PromptForInputsPage(InputGroup inputGroup, int stepNumber, int totalSteps
 
             int viewNumber = 1;
             bool anyRequired = false;
+
+            if (inputGroup.Inputs.Any(input => input.Display && input.GetView(viewNumber) != null && input.Required))
+            {
+                anyRequired = true;
+            }
```

Comment:
Reduce above to `bool anyRequired = inputGroup.Inputs.Any(input => input.Display && input.GetView(viewNumber) != null && input.Required)`

Also, not sure why the `input.GetView(viewNumber) != null` is needed.